test/core/store/fs/impl/test_https_netcdf.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2018-2024 by xcube team and contributors
+# Permissions are hereby granted under the terms of the MIT License:
+# https://opensource.org/licenses/MIT.
+
+import os.path
+import unittest
+
+import fsspec
+import rasterio as rio
+import rioxarray
+import s3fs
+import xarray
+import xarray as xr
+
+from xcube.core.store import new_data_store
+
+
+class HttpsNetcdfTest(unittest.TestCase):
+    """
+    This class tests a lazy access of a NetCDF file from a remote HTTPS server.
+    """
+
+    def test_open_netcdf_https(self):
+        """This test loads GAMIv2-0_2010-2020_100m.nc (217GB), which is available via
+        https://datapub.gfz-potsdam.de/download/10.5880.GFZ.1.4.2023.006-VEnuo/
+        """
+        fs_path = "download/10.5880.GFZ.1.4.2023.006-VEnuo/GAMIv2-0_2010-2020_100m.nc"
+        store = new_data_store("https", root="datapub.gfz-potsdam.de")
+        ds = store.open_data(fs_path, chunks={})
```

Comment:
This loads the data from remote source which is not wanted in a unittest. I would suggest using [pytest-recording](https://pypi.org/project/pytest-recording/), like in [xcube-stac](https://github.com/xcube-dev/xcube-stac?tab=readme-ov-file#some-notes-on-the-strategy-of-unit-testing-). This would mean that we add a new dependency pytest-recording to the test suite. Would that be okay? Or do you have a better suggestion?

Reply:
@forman `pytest-recording` does not recognize the request when opening the data with `xr.open_dataset`. Therefore no cassette is written and it just uses the actual https request. I don't think we can use `pytest-recording` in this case. Probably we still need to build a mock, right?

Reply:
I mocked the result of `xarray.open_dataset` now.

Reply:
Which is fine!